### PR TITLE
Migrate Reporting Notebooks Integration

### PR DESCRIPTION
**Pull request by davidcui-amzn**
### Description
* Add notebooks as a report source option in dashboards-reports plugin

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx [hunk 1]
```diff
@@ -681,7 +727,43 @@ export function ReportSettings(props: ReportSettingProps) {
       </div>
     );
 
+    const displayNotebooksSelect = 
+    reportSourceId === 'notebooksReportSource' ? (
+      (
```
Comment (joshuali925) [inline dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx:732]: why two parentheses?

File: dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx [hunk 2]
```diff
@@ -573,6 +608,17 @@ export function ReportSettings(props: ReportSettingProps) {
       .catch((error) => {
         console.log('error when fetching saved searches:', error);
       });
+
+    await httpClientProps
+      .get('/api/notebooks/')
```
Comment (joshuali925) [inline dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx:613]: `../api/notebooks/`? similar to how reporting api is called above

File: dashboards-reports/server/routes/utils/constants.ts [hunk 3]
```diff
@@ -167,6 +169,18 @@ export const GLOBAL_BASIC_COUNTER: CountersType = {
       },
     },
   },
+  notebook: {
+    pdf: {
+      download: {
+        count: 0,
```
Comment (joshuali925) [inline dashboards-reports/server/routes/utils/constants.ts:175]: are notebooks metrics recorded or just a place holder for now?

**Review by joshuali925 — APPROVED** [review]

File: dashboards-reports/server/routes/utils/constants.ts (hunk 3 above)
Comment (davidcui-amzn) [reply dashboards-reports/server/routes/utils/constants.ts:175]: This is a placeholder for now, notebooks metrics will need to be added

File: dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx (hunk 2 above)
Comment (davidcui-amzn) [reply dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx:613]: Good catch, will change

File: dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx (hunk 1 above)
Comment (davidcui-amzn) [reply dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx:732]: No reason, good catch. I can remove the inner set

**Review by zhongnansu — APPROVED** [review]

**Comment by davidcui-amzn** [comment]
Delaying merge on this PR until after `dev` branch is merged into `main` for the OpenSearch beta release

**Pull request merged**